ui/components/ui/account-list/account-list.js

```diff
@@ -0,0 +1,175 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+import { useI18nContext } from '../../../hooks/useI18nContext';
+import CheckBox, { CHECKED, INDETERMINATE, UNCHECKED } from '../check-box';
+import Identicon from '../identicon';
+import UserPreferencedCurrencyDisplay from '../../app/user-preferenced-currency-display';
+import { PRIMARY } from '../../../helpers/constants/common';
+import Tooltip from '../tooltip';
+
+const AccountList = ({
+  selectNewAccountViaModal,
+  accounts,
+  addressLastConnectedMap,
+  selectedAccounts,
+  nativeCurrency,
+  allAreSelected,
+  deselectAll,
+  selectAll,
+  handleAccountClick,
+}) => {
+  const t = useI18nContext();
+
+  const Header = () => {
+    let checked;
+    if (allAreSelected()) {
+      checked = CHECKED;
+    } else if (selectedAccounts.size === 0) {
+      checked = UNCHECKED;
+    } else {
+      checked = INDETERMINATE;
+    }
+
+    return (
+      <div
+        className={classnames({
+          'choose-account-list__header--one-item': accounts.length === 1,
+          'choose-account-list__header--two-items': accounts.length > 1,
```

Comment:
Should we use `multiple` instead of 2?  Not important just a thought.

Reply:
I agree, I'll change that!